File: epi_judge_java/epi/Anagrams.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.EpiTestComparator;
import epi.test_framework.GenericTest;
import epi.test_framework.LexicographicalListComparator;

import java.util.*;

public class Anagrams {

    @EpiTest(testDataFile = "anagrams.tsv")
    public static List<List<String>> findAnagrams(List<String> dictionary) {
        Map<String, List<String>> sortedToAnagramsMap = new HashMap<>();
        for (String s : dictionary) {
            var chars = s.toCharArray();
            Arrays.sort(chars);
            var sortedStr = new String(chars);
            sortedToAnagramsMap.computeIfAbsent(sortedStr, k -> new ArrayList<>()).add(s);
        }
        List<List<String>> ans = new ArrayList<>();
        for (var entry : sortedToAnagramsMap.entrySet()) {
            if (entry.getValue().size() == 1) continue;
            ans.add(entry.getValue());
        }
        return ans;
    }

    @EpiTestComparator
    public static boolean comp(List<List<String>> expected,
                               List<List<String>> result) {
        if (result == null) {
            return false;
        }
        for (List<String> l : expected) {
            Collections.sort(l);
        }
        expected.sort(new LexicographicalListComparator<>());
        for (List<String> l : result) {
            Collections.sort(l);
        }
        result.sort(new LexicographicalListComparator<>());
        return expected.equals(result);
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "Anagrams.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}

File: epi_judge_cpp/anagrams.cc
#include <string>
#include <vector>

#include "test_framework/generic_test.h"

using namespace std;

vector<vector<string>> FindAnagrams(const vector<string> &dictionary) {
    map<string, vector<string>> sorted_string_to_strings_map;
    for (const auto &s: dictionary) {
        string copy = s;
        std::sort(copy.begin(), copy.end());
        sorted_string_to_strings_map[copy].push_back(s);
    }
    vector<vector<string>> ans;
    for (auto &[_, strings]: sorted_string_to_strings_map) {
        if (strings.size() == 1) continue;
        ans.push_back(strings);
    }
    return ans;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"dictionary"};
    return GenericTestMain(args, "anagrams.cc", "anagrams.tsv", &FindAnagrams,
                           UnorderedComparator{}, param_names);
}
